app/src/main/java/org/metabrainz/mobile/presentation/features/barcode/BarcodeActivity.java

```diff
@@ -0,0 +1,87 @@
+package org.metabrainz.mobile.presentation.features.barcode;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+import org.metabrainz.mobile.R;
+
+import me.dm7.barcodescanner.zbar.Result;
+import me.dm7.barcodescanner.zbar.ZBarScannerView;
+
+public class BarcodeActivity extends Activity implements ZBarScannerView.ResultHandler {
```

Comment:
This is why I suggested yesterday to hide the ZBarScanner implementation behind an abstract class.

As ZBar is not updated frequently, we should hide 3rd party libraries implementation behind abstract classes. That way the **Activities do not implement the 3rd party directly**, but an abstract class that acts as the intermediary.

If we change the 3rd party plugin, all the methods remain untouched in the Activities (easier to change)

It's Clean Architecture good practice. I encourage you to try some exercise about this, as it might be useful to your professional career.

In pseudocode to give you an idea:
1. `abstract class BarcodeScan`: defined methods to scan things, and return results
2. `class BarcodeScanWithZbar implements BarcodeScan ZBarScannerView.ResultHandler`: implements the abstract class methods, and executes ZBar to get the results.
3. `BarcodeActivity implements BarcodeScanWithZbar`: instead of executing the 3rd party methods directly, executed the ones we've defined in the abstract class. If tomorrow we change to XXXScanner, we should only change `BarcodeActivity implements BarcodeScanWithXXXScanner`, but not how we scan things (same methods, defined by the abstract class)